dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/trackedentity/TrackerOwnershipManagerTest.java

```diff
@@ -506,6 +514,19 @@ void shouldFindTrackedEntityWhenTransferredToInaccessibleOrgUnitIfSuperUser()
     assertContainsOnly(List.of(trackedEntityA1.getUid()), trackedEntities);
   }
 
+  @Test
+  void shouldNotTransferOwnershipWhenOrgUnitNotAssociatedToProgram() {
+    OrganisationUnit notAssociatedOrgUnit = createOrganisationUnit('C');
+    organisationUnitService.addOrganisationUnit(notAssociatedOrgUnit);
+    Exception exception =
+        assertThrows(
+            AccessDeniedException.class,
+            () -> transferOwnership(trackedEntityA1, programA, notAssociatedOrgUnit));
+    assertEquals(
+        "User does not have access to change ownership for the entity-program combination",
```

Comment:
This sounds like an access error, which is true if the hasAccess() method fails.

Does it make sense to have a different error along the lines of "The program is not associated/supported in this org unit" and have that return specifically when the new validation you added fails?

Reply:
Yes, it probably makes more sense to return the error you mention, I'll change it.